### PR TITLE
Add drawing function 

### PR DESCRIPTION
**Pull request by ppeeou**
#### What this PR does / why we need it?

Code editor provides users with drawing tools

Currently, only the line drawing part is provided
-Add board component
-Add function to draw lines on a canvas

#### What are the relevant tickets?

Fixes #

### Checklist
- [ ] Added relevant tests or not required
- [ ] Didn't break anything

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "noEmit": true,
     "jsx": "react",
     "allowJs": true,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+    "downlevelIteration": true
```
Comment (ppeeou) [inline tsconfig.json:23]: I added it because of the [code below](https://github.com/yorkie-team/yorkie-codepair/pull/45/files#diff-75a5cbb43d1dd466fc40f917296f59b28236ee256bbd1301fb3579b123f3bdeeR132-R134
)
We should turn on this option when we want to iterate over an array or string type on Typescript
```js
for (const shape of shapes) {
    this.draw(shape, canvas);
}
```

If we use this option, the bundle size increases, but I used it for readability.🤔

https://mariusschulz.com/blog/downlevel-iteration-for-es3-es5-in-typescript
In conclusion, the bundler file grows to serve the emoji.

**Review by hackerwins — CHANGES_REQUESTED** [review]
@ppeeou 
Drawing component's icons appear weird in dark mode.

Screen Shot 2021-02-21 at 11 51 43

**Comment by ppeeou** [comment]
Could I ask which part is weird??
The drawing is selected and it is marked in blue.

In the dark mode, I think the pencil color should be white🤔

**Comment by hackerwins** [comment]
@ppeeou 
Looks like the style is broken in dark mode. How about displaying the pencil button in white like other icons, and drawing lines in white?

**Comment by ppeeou** [comment]
@hackerwins apply to it !:) 
The material design makes it easy to apply dark mode

image

File: src/index.css [hunk 2]
```diff
@@ -9,3 +9,101 @@ body {
...
+}
+.sw-resize {
+  cursor: sw-resize;
+}
+.nesw-resize {
+  cursor: nesw-resize;
+}
+.nwse-resize {
+  cursor: nwse-resize;
+}
```
Comment (hackerwins) [inline src/index.css:109]: It seems that the mouse css property is too open. What if we limit these properties to be reflected under the canvas?

File: src/components/Editor/Board/Canvas/Shape.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// TODO yorkie-sdk-js ArrayProxy
```
Comment (hackerwins) [inline src/components/Editor/Board/Canvas/Shape.ts:1]: To remember the creator and responsibilities of TODO, please leave a TODO comment in the pattern below.

```suggestion
// TODO(ppeeou): yorkie-sdk-js ArrayProxy
```

**Review by hackerwins — APPROVED** [review]
Thank you for your work. I left a brief comment.

**Comment by ppeeou** [comment]
@hackerwins Thank you for your comment
I applied to it
I think I can get good habits thanks to you.

**Comment by hackerwins** [comment]
@ppeeou
Additionally, when the drawing board is visible, unintended scrolling is created. How about fixing it before merging?
Screen Shot 2021-02-24 at 12 01 46
